src/transformers/training_args.py

```diff
@@ -2056,6 +2062,13 @@ def __post_init__(self):
                 "This is not supported and we recommend you to update your version."
             )
 
+        if self.include_inputs_for_metrics:
+            warnings.warn(
+                "Using `include_inputs_for_metrics` is deprecated and will be removed in version 5 of 🤗 Transformers. Please use `include_for_metrics` list argument instead.",
+                FutureWarning,
+            )
```

Comment:
We prefer to use logger.warning as users will have more control over their logs 

```suggestion
            logger.warning(
                "Using `include_inputs_for_metrics` is deprecated and will be removed in version 5 of 🤗 Transformers. Please use `include_for_metrics` list argument instead."
            )
```